### ISSUE TITLE
Gnome File Sharing doesn't work

### ISSUE THREAD

**Issue by Neon-44** [NixOS/nixpkgs#234003]
### Describe the bug
Gnomes Network sharing doesn't work

can't access the "Public" Folder from other Systems when activating Sharing in GNOME Settings

### Steps To Reproduce
1. open gnome settings and enable file sharing
2. enter the link on another system
3. doesn't work

### Expected behavior
i am supposed to be able to access the public folder from other devices in the same LAN

### Notify maintainers

will in the comments once i find out who it is

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

**Comment by TheOddler** [comment]
Did you ever figure this out? I'm in the same boat at the moment.

**Comment by Neon-44** [comment]
Nope, am Sorry. I used Rsync at first and then set up a NAS.

**Comment by TheOddler** [comment]
Fair enough, setting up a NAS will happen here some day too, for now it's just an external driver I move around :sweat_smile:

**Comment by RSWilli** [comment]
I just ran into this as well, and after some debugging I got it working. With the help of:

* https://discourse.gnome.org/t/help-what-is-the-port-range-that-gnome-user-share-uses/16793
* https://gitlab.gnome.org/GNOME/gnome-user-share/-/work_items/25
* https://gitlab.gnome.org/GNOME/gnome-user-share/-/work_items/28

Reasons why this didn't work:

1. The firewall blocks the port. Check if you have `networking.firewall.enable = true` (which is the default)
2. gnome user share chooses a random free port. This means you either need to disable the firewall or permit a whole range of ports (not quite sure what the actual range is).
3. The gnome UI didn't show me the port as part of the address, only the hostname. In my case it showed `dav://think` but I needed `dav://think:40561` on the other machine to access the files. I found this port using `ss` / `nmap`.

CC @bobby285271 @dasj19 @hedning @jtojnar
